crates/analysis/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+use sightglass_data::{Measurement, Phase, Summary};
+use std::{borrow::Cow, collections::HashSet};
+
+/// Summarize measurements grouped by: architecture, engine, benchmark file, phase and event.
+pub fn aggregate<'a>(measurements: &[Measurement<'a>]) -> Vec<Summary<'a>> {
+    let mut summaries = Vec::new();
+    for (arch, engine, wasm, phase, event) in keys(&measurements) {
+        let grouped_counts: Vec<_> = measurements
+            .iter()
+            .filter(|m| m.phase == phase && m.event == event)
+            .map(|m| m.count)
+            .collect();
+        summaries.push(Summary {
+            arch,
+            engine,
+            wasm,
+            phase,
+            event,
+            min: grouped_counts
+                .iter()
+                .cloned()
+                .min()
+                .expect("at least one element"),
+            max: grouped_counts
+                .iter()
+                .cloned()
+                .max()
+                .expect("at least one element"),
+            mean: mean(&grouped_counts),
+            deviation: deviation(&grouped_counts),
+        })
+    }
+    summaries
+}
+
+/// Extract the keys (i.e. `phase` and `event`) for the groups of measurements to aggregate.
+fn keys<'a>(measurements: &[Measurement<'a>]) -> Vec<Key<'a>> {
+    let set: HashSet<_> = measurements
+        .iter()
+        .cloned()
+        .map(|m| (m.arch, m.engine, m.wasm, m.phase, m.event))
+        .collect();
+    set.into_iter().collect()
+}
+type Key<'a> = (
```

Comment:
It probably makes sense to just define a proper `struct` for this, since we have to explain what "key" means in the doc comment but then also its definition is too big to do inline, so we have a type alias. Instead, if we just define a proper `struct`, we can get rid of the mini-explanation of what a key is in the doc comment and we have a single place where we need to explain what it is.